### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by fsalata** [ahmed93/AAShimmerView#1]
I'm experiencing high CPU usage even after stopping the shimmer effect.

**Comment by theappedia** [comment]
Same problem, crazy high CPU usage!

So i've noticed... the shimmer view works fine until **_either_** of the two things change:
1) If you stop shimmer effect (the CPU usage peaks (and stays there) as soon as the effect is stopped).

2) If you go onto a different View altogether without stopping the shimmer effect the CPU peaks again (and stays there). However, as soon as you come back to the View that is displaying the effect (basically the view that contains the shimmerView) the CPU usage gets back to normal.

In conclusion, once you initiate the effect, **_unless_** the shimmer view is being displayed on screen, there is really high CPU usage.

Lovely effect but unable to use coz of the above problem :( hope it can be fixed soon.

**Comment by ahmed93** [comment]
Thanks, guys showing me this issue, i just noticed the issue you are talking about. 
Is the issue with my approach or this approach shouldn't cause this crazy CPU usage ??

Any ideas are more than welcome

**Comment by ahmed93** [comment]
if the issue re-appeared plz reopen again
